### PR TITLE
Fast io cal

### PR DESCRIPTION
**Pull request by AaronParsons**
Adds io.read_hera_calfits for quick reading of calfits files.

### CODE REVIEW

**Review by jsdillon — CHANGES_REQUESTED** [review]
Nice stuff! I've got a handful of relatively minor change requests.

Two other things: 
1) It'd be nice to show some example (screenshots) of the speedups involved here.
2) `read_hera_calfits()` is a bit light on inline comments

File: hera_cal/io.py [hunk 1]
```diff
@@ -1150,6 +1151,156 @@ def _make_datacontainer(self, rv, key='data'):
...
+                else:
+                    info['pols'] = set(_pols)
+                inds[_jhash] = {pol: idx for idx, pol in enumerate(_pols)}
+            inds[filename] = (inds[_ahash], inds[_jhash])
+            if cnt == 0:
+                if 'ANTXYZ' in antdata.names:
+                    info['antpos'] = antdata["ANTXYZ"]
+                info['freqs'] = uvutils._fits_gethduaxis(fname[0], 4)
+            if check:
+                gain_convention = hdr.pop("GNCONVEN")
```
[the remark this answers is not in the document]
Comment (AaronParsons) [reply hera_cal/io.py:1223]: Done

File: hera_cal/io.py [hunk 2]
```diff
@@ -1150,6 +1151,156 @@ def _make_datacontainer(self, rv, key='data'):
...
+            if _ahash not in inds:
+                inds[_ahash] = {ant: idx for idx, ant in enumerate(_ants)}
+                if 'ants' in info:
+                    info['ants'].intersection_update(set(inds[_ahash].keys()))
+                else:
+                    info['ants'] = set(inds[_ahash].keys())
+            jones_array = uvutils._fits_gethduaxis(fname[0], 2)
+            _jhash = hash(jones_array.tobytes())
+            if _jhash not in inds:
+                x_orient = hdr['XORIENT']
```
[the remark this answers is not in the document]
Comment (AaronParsons) [reply hera_cal/io.py:1209]: Done

File: hera_cal/io.py [hunk 3]
```diff
@@ -1150,6 +1151,156 @@ def _make_datacontainer(self, rv, key='data'):
         return dc
 
 
+def read_hera_calfits(filenames, ants=None, pols=None,
```
[the remark this answers is not in the document]
Comment (AaronParsons) [reply hera_cal/io.py:1154]: Done

File: hera_cal/tests/test_io.py [hunk 4]
```diff
@@ -816,6 +816,156 @@ def test_errors(self):
...
+        # test select on pols
+        rv1 = io.read_hera_calfits(self.fname_xx)
+        rv2 = io.read_hera_calfits(self.fname_2pol, pols=['Jee'])
+        for k in rv2['gains'].keys():
+            assert k[1] == 'Jee'
+            assert k in rv1['gains']
+            np.testing.assert_array_equal(rv1['gains'][k], rv2['gains'][k])
+
+
+class Test_ReadHeraCalfits(object):
```
[the remark this answers is not in the document]
Comment (AaronParsons) [reply hera_cal/tests/test_io.py:896]: urk

File: hera_cal/tests/test_io.py [hunk 5]
```diff
@@ -816,6 +816,156 @@ def test_errors(self):
...
+            assert qual.shape == shape
+        assert rv['info']['pols'] == set(['Jnn', 'Jee'])
+
+        # test list loading
+        rv = io.read_hera_calfits([self.fname_xx, self.fname_yy], read_gains=True, read_flags=True,
+                                  read_quality=True, read_tot_quality=False,
+                                  check=True, verbose=True)
+        for key in ('gains', 'flags', 'quality'):
+            assert len(rv[key].keys()) == 36
+
```
[the remark this answers is not in the document]
Comment (AaronParsons) [reply hera_cal/tests/test_io.py:872]: Done.

File: hera_cal/io.py [hunk 6]
```diff
@@ -1150,6 +1151,156 @@ def _make_datacontainer(self, rv, key='data'):
...
+                      read_gains=True, read_flags=False, read_quality=False, read_tot_quality=False,
+                      check=False, dtype=np.complex128, verbose=False):
+    '''A faster interface to getting data out of HERA calfits files. Only concatenates
+    along time axis. Puts times in ascending order,
+    but does not check that files are contiguous.
```
[the remark this answers is not in the document]
Comment (AaronParsons) [reply hera_cal/io.py:1159]: Done.

**Review by jsdillon — APPROVED** [review]

**Pull request merged**